src/sketchup-stl/exporter.rb

```diff
@@ -94,11 +94,22 @@ def self.find_faces(file, entities, facet_count, scale, tform)
         facet_count
       end
 
+      # GitHub Issue #162: Binary preview is black in Mac OS.
+      # Caused because normals were not being requested in the Face#mesh call,
+      # and so not being exported.
+      # face.mesh was being called using only 0. We need to request normals
+      # using face.mesh(4)
+      # Flags for Face#mesh(flags)
+      #   0: Include PolygonMeshPoints
+      #   1: Include PolygonMeshUVQFront
+      #   2: Include PolygonMeshUVQBack
+      #   4: Include PolygonMeshNormals
+
       def self.write_face(file, face, scale, tform)
         normal = face.normal
         normal.transform!(tform)
         normal.normalize!
-        mesh = face.mesh(0)
+        mesh = face.mesh(0 | 4)
```

Comment:
I'm curious though, `write_face_binary` takes the vertex normal from the mesh. Even though the method is given a normal. Also, the normal we pass on has been processed to match the nested transformation currently in play.

See `write_face`
https://github.com/SketchUp/sketchup-stl/blob/master/src/sketchup-stl/exporter.rb#L97

We compute a global normal and pass that to `write_face_binary`
https://github.com/SketchUp/sketchup-stl/blob/master/src/sketchup-stl/exporter.rb#L133

But instead of using it we used the an arbitrary vertex normal. One potential issue here is that the normal we use isn't transformed to the global space. But also that we use a vertex-normal - assuming it's the same direction as the face. If an edge is soft/smooth then the vertex normals connected to that face will not be the same direction as the face.

I think the correct fix here is to use the normal we extract from `face.normal` - similar to what we do in `write_face_ascii`.